handsontable/src/plugins/loading/loading.js

```diff
@@ -0,0 +1,286 @@
+
+export const PLUGIN_KEY = 'loading';
+export const PLUGIN_PRIORITY = 350;
+export const LOADING_CLASS_NAME = `ht-${PLUGIN_KEY}`;
+
+/**
+ * @plugin Loading
+ * @class Loading
+ *
+ * @description
+ * The loading plugin provides a loading overlay system for Handsontable using the Dialog plugin.
+ * It displays a loading indicator with customizable title, icon, and description.
+ *
+ * In order to enable the loading mechanism, {@link Options#loading} option must be set to `true`.
+ *
+ * The plugin provides several configuration options to customize the loading behavior and appearance:
+ * - `icon`: Loading icon to display HTML (as string) in svg format (default: `<svg ... />`).
+ * - `title`: Loading title to display (default: 'Loading...').
+ * - `description`: Loading description to display (default: '').
+ *
+ * @example
+ *
+ * ::: only-for javascript
+ * ```js
+ * // Enable loading plugin with default options
+ * dialog: true, // Dialog plugin must be enabled
+ * loading: true,
+ *
+ * // Enable loading plugin with custom configuration
+ * dialog: true, // Dialog plugin must be enabled
+ * loading: {
+ *   icon: 'A custom loading icon in SVG format',
+ *   title: 'Custom loading title',
+ *   description: 'Custom loading description',
+ * }
+ *
+ * // Access to loading plugin instance:
+ * const loadingPlugin = hot.getPlugin('loading');
+ *
+ * // Show a loading programmatically:
+ * loadingPlugin.show();
+ *
+ * // Hide the loading programmatically:
+ * loadingPlugin.hide();
+ *
+ * // Check if dialog is visible:
+ * const isVisible = loadingPlugin.isVisible();
+ * ```
+ * :::
+ *
+ * ::: only-for react
+ * ```jsx
+ * const MyComponent = () => {
+ *   const hotRef = useRef(null);
+ *
+ *   useEffect(() => {
+ *     const hot = hotRef.current.hotInstance;
+ *     const loadingPlugin = hot.getPlugin('loading');
+ *
+ *     loadingPlugin.show();
+ *   }, []);
+ *
+ *   return (
+ *     <HotTable
+ *       ref={hotRef}
+ *       settings={{
+ *         data: data,
+ *         dialog: true, // Dialog plugin must be enabled
+ *         loading: {
+ *           icon: 'A custom loading icon in SVG format',
+ *           title: 'Custom loading title',
+ *           description: 'Custom loading description',
+ *         }
+ *       }}
+ *     />
+ *   );
+ * }
+ * ```
+ * :::
+ *
+ * ::: only-for angular
+ * ```ts
+ * hotSettings: Handsontable.GridSettings = {
+ *   data: data,
+ *   dialog: true, // Dialog plugin must be enabled
+ *   loading: {
+ *     icon: 'A custom loading icon in SVG format',
+ *     title: 'Custom loading title',
+ *     description: 'Custom loading description',
+ *   }
+ * }
+ * ```
+ *
+ * ```html
+ * <hot-table
+ *   [settings]="hotSettings">
+ * </hot-table>
+ * ```
+ * :::
+ */
+
+export class Loading extends BasePlugin {
+  static get PLUGIN_KEY() {
+    return PLUGIN_KEY;
+  }
+
+  static get PLUGIN_PRIORITY() {
+    return PLUGIN_PRIORITY;
+  }
+
+  static get DEFAULT_SETTINGS() {
```

Comment:
When some plugin logic depends on another plugin, it's necessary to add a static `PLUGIN_DEPS` field as follows:
```suggestion
  static get PLUGIN_PRIORITY() {
    return PLUGIN_PRIORITY;
  }

  static get PLUGIN_DEPS() {
    return [
      'plugin:Dialog',
    ];
  }

  static get DEFAULT_SETTINGS() {
```
Thanks to that, when someone registers only the Loading plugin in their app, there will be an exception about the missing `Dialog` plugin.